services/fib2/src/main.rs

```diff
@@ -33,6 +33,8 @@ extern "C" {
     #[polkavm_import(index = 6)]
     pub fn assign(c: u64, o: u64) -> u64;
 
+    #[polkavm_import(index = 8)]
+    pub fn checkpoint() -> u64;
```

Comment:
ADDED checkpoint so as to have panic still record writes -- this is an important fix!